### PR TITLE
fix parser to allow no storage to be passed to knowledge base

### PR DESCRIPTION
**Pull request by dusvyat**
fix parser to allow for user to not specify storage in creation of knowledge base

also see PR https://github.com/mindsdb/mindsdb/pull/7886

### CODE REVIEW

File: mindsdb_sql/parser/dialects/mindsdb/parser.py [hunk 1]
```diff
@@ -99,18 +101,20 @@ def create_kb(self, p):
...
         # check model and storage are in params
-        model = params.pop('model', None) or params.pop('MODEL', None)  # case insensitive
-        storage = params.pop('storage', None) or params.pop('STORAGE', None)  # case insensitive
+        params = {k.lower(): v for k, v in params.items()}  # case insensitive
+        model = params.pop('model', None)
+        storage = params.pop('storage', None)
+
+        if not storage:
+            # convert to identifier
+            storage = Identifier('_default._vector_store')
```
Comment (ea-rus) [inline mindsdb_sql/parser/dialects/mindsdb/parser.py:110]: I think it should be in mindsdb executor (KB controller) 
This name is not related to sql parsing

**Review by ea-rus — CHANGES_REQUESTED** [review]

File: mindsdb_sql/parser/dialects/mindsdb/parser.py [hunk 2]
```diff
@@ -99,18 +101,21 @@ def create_kb(self, p):
...
-        model = params.pop('model', None) or params.pop('MODEL', None)  # case insensitive
-        storage = params.pop('storage', None) or params.pop('STORAGE', None)  # case insensitive
+        params = {k.lower(): v for k, v in params.items()}  # case insensitive
+        model = params.pop('model', None)
+        storage = params.pop('storage', None)
+
+        if not storage:
+            # convert to identifier
+            if isinstance(model, str):
+                storage = Identifier(storage)
```
Comment (ea-rus) [inline mindsdb_sql/parser/dialects/mindsdb/parser.py:111]: I think something wrong here: if not storage: use it

Comment (dusvyat) [reply mindsdb_sql/parser/dialects/mindsdb/parser.py]: good, point. fixed
